One big sprite

XinYenFon opened the pull request.

XinYenFon committed:
One big sprite...
Signed-off-by: Antes <antes@simplemachines.org>

XinYenFon committed:
better sprite image (remove some unused space).
Signed-off-by: Antes <antes@simplemachines.org>

live627 commented on Themes/default/css/index.css:
```diff
@@ -1773,8 +1773,9 @@ tr.windowbg td, tr.windowbg2 td, .table_grid tr td {
 	background: url(../images/generic_icons.png) no-repeat 0 0;
 	vertical-align: middle;
 }
-.generic_icons.gender_None {
```
This is here so that a help icon won't appear as a gender icon when no gender was selected.

XinYenFon replied:
I know I added that, but afaik @MissAllSunday changed that code...

live627 replied:
Check Yoshi's test site. That code still applies.

XinYenFon replied:
okay, if its... I'll change things back and update the PR.

MissAllSunday replied:
NO, I didn't....

live627 commented on Sources/Subs-Menu.php:
```diff
@@ -123,10 +123,8 @@ function createMenu($menuData, $menuOptions = array())
 						// Does this area have its own icon?
 						if (!isset($area['force_menu_into_arms_of_another_menu']) && $user_info['name'] == 'iamanoompaloompa')
 							$menu_context['sections'][$section_id]['areas'][$area_id] = unserialize(base64_decode('YTozOntzOjU6ImxhYmVsIjtzOjEyOiJPb21wYSBMb29tcGEiO3M6MzoidXJsIjtzOjQzOiJodHRwOi8vZW4ud2lraXBlZGlhLm9yZy93aWtpL09vbXBhX0xvb21wYXM/IjtzOjQ6Imljb24iO3M6ODY6IjxpbWcgc3JjPSJodHRwOi8vd3d3LnNpbXBsZW1hY2hpbmVzLm9yZy9pbWFnZXMvb29tcGEuZ2lmIiBhbHQ9IkknbSBhbiBPb21wYSBMb29tcGEiIC8+Ijt9'));
-						elseif (isset($area['icon']) && file_exists($settings['theme_dir'] . '/images/admin/' . $area['icon']))
```
This is for compatibility with old mods.

XinYenFon replied:
I'll revert that my bad, i forgot...

XinYenFon committed:
Conflict resolved (?) :P
- reverted gender_None
- reverted compatibility with fix of fallback to CSS.

Signed-off-by: Antes <antes@simplemachines.org>

XinYenFon committed:
Why always :( do ... hard ... way...

Merge branch 'release-2.1' of git://github.com/simplemachines/SMF2.1 into theme_fix

# By Illori (3) and others
# Via Michael Eshom (7) and John Rayes (3)
* 'release-2.1' of git://github.com/simplemachines/SMF2.1:
  ! Remove debugging code and fix a typo
  ! Typo (fixes #2146)
  Duplicated class .generic_icons.warning_watch
  ! duplicate name in the check array
  ! The CreatePost-Notify background task expects "msg" for both posts and topics
  remove modifykarma
  last fix for #2120
  ! Harmonise alignment for admin quick search boxes
  ! Don't show enoticon button when no smileys present (#2128)
  fix touch

Conflicts:
	Themes/default/css/index.css

Signed-off-by: Antes <antes@simplemachines.org>
